### PR TITLE
PIM-6338: add behats for product model enrich

### PR DESCRIPTION
**Pull request by grena**
This PR adds first behats for the enrich of product models.

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | N
| Added Behats                      | Y
| Added integration tests           | N
| Changelog updated                 | N
| Review and 2 GTM                  | Todo

### CODE REVIEW

File: features/product-model/edit_product_model.feature [hunk 1]
```diff
@@ -0,0 +1,39 @@
+@javascript
+Feature: Edit a product model
+  In order to enrich the catalog
+  As a regular user
+  I need to be able edit and save a product model
+
+  Background:
+    Given a "catalog_modeling" catalog configuration
+    And the following root product models:
```
Comment (jjanvier) [inline features/product-model/edit_product_model.feature]: now there are product models on the catalog

Comment (grena) [inline features/product-model/edit_product_model.feature]: Yup, done 👍

File: features/Context/FixturesContext.php [hunk 2]
```diff
@@ -1515,6 +1517,24 @@ public function getProduct($sku)
...
     /**
+     * @param string $code
+     *
+     * @throws \InvalidArgumentException
+     *
+     * @return ProductModelInterface
+     */
+    public function getProductModel(string $code): ProductModelInterface
+    {
+        $productModel = $this->spin(function () use ($code) {
```
Comment (SamirBoulil) [inline features/Context/FixturesContext.php]: Do you really need to spin when calling a repository ?

Comment (grena) [inline features/Context/FixturesContext.php]: TBH I don't know, it's done this way for product, and I think there is a reason :(

Comment (aRn0D) [inline features/Context/FixturesContext.php]: You don't need to that in this context!

File: features/Pim/Behat/Context/Domain/Enrich/ProductModelContext.php [hunk 3]
```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace Pim\Behat\Context\Domain\Enrich;
+
+use Context\Spin\SpinCapableTrait;
+use Context\Traits\ClosestTrait;
+use Pim\Behat\Context\PimContext;
+
+class ProductModelContext extends PimContext
```
Comment (aRn0D) [inline features/Pim/Behat/Context/Domain/Enrich/ProductModelContext.php]: You don't need to extend this context, you can either implement `Context` interface or extends `RawMinkContext`

Comment (aRn0D) [inline features/Pim/Behat/Context/Domain/Enrich/ProductModelContext.php]: You need to create a constructor and inject the dependency you need, you should have a look to `Pim\Behat\Context\Storage\ProductModelStorage`

**Pull request merged**